src/main/java/com/ibm/jesseg/prometheus/SQLMetricPopulator.java

```diff
@@ -149,7 +149,7 @@ private void gatherData() throws SQLException {
               }
             }
             double value = rs.getDouble(i);
-            gauge.set(value);

```

Comment:
I haven't fully wrapped my head around the Prometheus APIs. This seems correct, but I'm cautious because the `labels()` function returns a new child of the original Gauge object. Not sure what implications that has. Would it be better to do this upon construction of the object? For instance (untested/unverified code):

```java
    Gauge ret = Gauge.build()
        .name(_gaugeName).help(_help).register();
        .name(_gaugeName)
        .help(_help)
        .labelNames("hostname")
        .labels(m_config.getHostNameForDisplay().replaceAll("\\..*", ""))
        .register();
    m_gauges.put(_gaugeName, ret);
```

Reply:
From my understanding `Gauge.Child` is the object that holds the metrics value. There isn't a way to set the label value during the construction of the metric like you suggested due to type incompatibility.  We could set label during creation in `getGauge()` and store the Child object instead on the Gauge object. We would need to update the indexing logic in in 'gatherData()'. Not sure if this is any more correct.